ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/cancellations/CancellationTable.tsx

```diff
@@ -0,0 +1,101 @@
+'use client';
+
+import { useRef } from 'react';
+import { NewButton } from '@inngest/components/Button';
+import { Table } from '@inngest/components/Table';
+import { Time } from '@inngest/components/Time';
+import { createColumnHelper, getCoreRowModel } from '@tanstack/react-table';
+
+import { useCancellations } from './useCancellations';
+
+type Cancellation = {
+  createdAt: string;
+  id: string;
+  queuedAtMax: string;
+  queuedAtMin: string | null;
+};
+
+type Props = {
+  envSlug: string;
+  fnSlug: string;
+};
+
+export function CancellationTable({ envSlug, fnSlug }: Props) {
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+
+  const {
+    data: items,
+    fetchNextPage,
+    hasNextPage,
+    isFetching,
+    isInitiallyFetching,
+  } = useCancellations({ envSlug, fnSlug });
+
+  let blankSlate = <p>No results</p>;
+  if (isInitiallyFetching) {
+    blankSlate = <p>Loading...</p>;
+  }
+
+  return (
+    <div className="flex flex-col items-center">
+      <div className="mb-8 self-stretch">
+        <Table
+          blankState={blankSlate}
+          options={{
+            columns,
+            data: items,
+            enableSorting: false,
+            getCoreRowModel: getCoreRowModel(),
+          }}
+          tableContainerRef={tableContainerRef}
+        />
+      </div>
+
+      {!isInitiallyFetching && (
+        <span>
+          <NewButton
+            appearance="outlined"
+            disabled={isFetching || !hasNextPage}
+            label="Load More"
+            onClick={() => fetchNextPage()}
+          />
+        </span>
+      )}
+    </div>
+  );
+}
+
+const columnHelper = createColumnHelper<Cancellation>();
+
+// TODO: Add column for cancellation deletion
+const columns = [
+  columnHelper.accessor('createdAt', {
+    header: () => <span>Created at</span>,
+    cell: (props) => {
+      return <Time value={props.getValue()} />;
```

Comment:
All these cells should be using the Cell components https://github.com/inngest/inngest/blob/main/ui/packages/components/src/Table/Cell.tsx